cleanlab_pinecone_rag_workflow/Cleanlab_Pinecone_RAG.ipynb

```diff
@@ -0,0 +1,2673 @@
+    "        query_embedding = self.model.encode(query)\n",
+    "        \n",
+    "        try:\n",
+    "            results = self.index.query(\n",
+    "                vector=query_embedding.tolist(),\n",
+    "                top_k=top_k,\n",
+    "                filter=filter_query,\n",
+    "                include_metadata=True\n",
+    "            )\n",
+    "            \n",
+    "            return [(match.metadata['text'], {k: v for k, v in match.metadata.items() if k != 'text'}) for match in results.matches]\n",
+    "        except Exception as e:\n",
+    "            print(f\"Error during search: {e}\")\n",
+    "            return []\n",
+    "\n",
+    "    def delete_index(self) -> None:\n",
+    "        \"\"\"\n",
+    "        Delete the Pinecone index.\n",
+    "        \n",
+    "        Raises:\n",
+    "            Exception: If there's an error during the deletion process.\n",
+    "        \"\"\"\n",
+    "        try:\n",
+    "            self.pc.delete_index(self.index_name)\n",
+    "            print(f\"Index '{self.index_name}' has been deleted.\")\n",
+    "        except Exception as e:\n",
+    "            print(f\"Error deleting index: {e}\")\n",
+    "\n",
+    "    def extract_chunks_from_index(self) -> pd.DataFrame:\n",
+    "        \"\"\"\n",
+    "        Extract document chunks and metadata from the Pinecone index into a DataFrame.\n",
+    "        \n",
+    "        Returns:\n",
+    "            pd.DataFrame: DataFrame containing chunk data and metadata.\n",
+    "                          Columns include all metadata fields stored in the index.\n",
+    "        \n",
+    "        Raises:\n",
+    "            Exception: If there's an error retrieving chunks from the index.\n",
+    "        \"\"\"\n",
+    "        stats = self.index.describe_index_stats()\n",
+    "        total_vectors = stats.total_vector_count\n",
+    "        dimension = stats.dimension\n",
+    "        \n",
+    "        print(f\"Index name: {self.index_name}\")\n",
+    "        print(f\"Total vectors according to stats: {total_vectors}\")\n",
+    "        print(f\"Vector dimension: {dimension}\")\n",
+    "\n",
+    "        try:\n",
+    "            results = self.index.query(\n",
+    "                vector=[0.0] * dimension,\n",
+    "                top_k=10000,\n",
+    "                include_values=False,\n",
+    "                include_metadata=True\n",
+    "            )\n",
+    "\n",
+    "            chunk_data = [match.metadata for match in results.matches]\n",
+    "            chunk_df = pd.DataFrame(chunk_data)\n",
+    "\n",
+    "            print(f\"Total chunks retrieved: {len(chunk_df)}\")\n",
+    "            return chunk_df\n",
+    "\n",
+    "        except Exception as e:\n",
+    "            print(f\"Error retrieving chunks from index: {e}\")\n",
+    "            return pd.DataFrame()\n",
+    "\n",
+    "    def delete_chunks(self, chunk_ids: List[str]) -> None:\n",
```

Comment:
Nit This function feels unnecessary since you're just running an operation and passing up an exception if it has an exception